DevelopGitHub/GitHubExtension/GitHubExtension.Templates/Commands/TemplatesCommand.cs

```diff
@@ -0,0 +1,132 @@
+             string repositoryName, string pathToFile, string token)
+        {
+            var issueTemplateUri =
+                GetTemplatesRequestUriWithToken(repositoryName, pathToFile, token);
+            var requestMessage = new HttpRequestMessage(HttpMethod.Put, issueTemplateUri).CreateTemplateMessage(message, content);
+            var response = await _httpClient.GetResponse(requestMessage);
+
+            return response;
+        }
+
+        public async Task<HttpResponseMessage> UpdatePullRequestTemplateAsync(string message, string content,
+            string repositoryName, string pathToFile, string token, string sha)
+        {
+            var pullRequestTemplate =
+                GetTemplatesRequestUriWithToken(repositoryName, pathToFile, token);
+            var requestMessage = new HttpRequestMessage(HttpMethod.Put, pullRequestTemplate).UpdateTemplateMessage(message, content, sha);
+            var response = await _httpClient.GetResponse(requestMessage);
+
+            return response;
+        }
+
+        public async Task<HttpResponseMessage> UpdateIssueTemplateAsync(string message, string content,
+             string repositoryName, string pathToFile, string token, string sha)
+        {
+            var issueTemplateUri =
+                GetTemplatesRequestUriWithToken( repositoryName, pathToFile, token);
+            var requestMessage = new HttpRequestMessage(HttpMethod.Put, issueTemplateUri).UpdateTemplateMessage(message, content, sha);
+            var response = await _httpClient.GetResponse(requestMessage);
+
+            return response;
+        }
+
+        public async Task<bool> IsOwner(string userName, string repositoryName)
+        {
+            var ownerUri = GetOwnerUri(repositoryName);
+            var requestMessage = new HttpRequestMessage(HttpMethod.Get, ownerUri).AddHeadersForGitHub();
+            var response = await _httpClient.GetResponse(requestMessage);
+
+            var messageContent = await response.Content.ReadAsStringAsync();
+            var model = JsonConvert.DeserializeObject<RootObject>(messageContent);
+            var login = model.owner.login;
+
+            if (login == userName)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        public async Task<string> GetShaPullRequestTemplate( string repositoryName, string pathToFile)
+        {
+            var pullRequestTemplateUri = GetTemplatesRequestUri( repositoryName, pathToFile);
+            var message = new HttpRequestMessage(HttpMethod.Get, pullRequestTemplateUri).AddHeadersForGitHub();
+            var response = await _httpClient.GetResponse(message);
+
+            var messageContent = await response.Content.ReadAsStringAsync();
+            var model = JsonConvert.DeserializeObject<GitHubTemplatesModel>(messageContent);
+            var sha = model.Sha;
+
+            return sha;
```

Comment:
We need it to update a file. 

PUT /repos/:owner/:repo/contents/:path
Required. The blob SHA of the file being replaced.